File: csharp/RockPaperScissors/RockPaperScissors.cs
using System.Xml.Schema;
using System.ComponentModel.DataAnnotations;
namespace RockPaperScissors;

public class RockPaperScissors
{
    public RockPaperScissors(string playerOne, string playerTwo)
    {
        var valid = new List<string>() { "rock", "paper", "scissors" };
        if (!valid.Contains(playerOne))
        {
            throw new ArgumentException($"invalid move {playerOne}");
        }
        if (!valid.Contains(playerTwo))
        {
            throw new ArgumentException($"invalid move {playerTwo}");
        }
        this.PlayerOne = playerOne;
        this.PlayerTwo = playerTwo;
    }

    public string PlayerOne { get; }
    public string PlayerTwo { get; }

    public int Winner()
    {
        if (this.PlayerOne.Equals(this.PlayerTwo))
        {
            return 0;
        }
        if (this.PlayerOne.Equals("rock") && this.PlayerTwo.Equals("scissors"))
        {
            return -1;
        }
        if (this.PlayerOne.Equals("scissors") && this.PlayerTwo.Equals("rock"))
        {
            return 1;
        }
        if (this.PlayerOne.Equals("paper") && this.PlayerTwo.Equals("rock"))
        {
            return -1;
        }
        if (this.PlayerOne.Equals("rock") && this.PlayerTwo.Equals("paper"))
        {
            return 1;
        }
        if (this.PlayerOne.Equals("scissors") && this.PlayerTwo.Equals("paper"))
        {
            return -1;
        }
        if (this.PlayerOne.Equals("paper") && this.PlayerTwo.Equals("scissors"))
        {
            return 1;
        }
        throw new ArgumentException($"{this.PlayerOne}:{this.PlayerTwo} is undefined");
    }
}

File: python/rock_paper_scissors/rock_paper_scissors.py
class RockPaperScissors:
    def __init__(self, player_one: str, player_two: str):
        valid = ["rock", "paper", "scissors"]
        if player_one not in valid:
            raise ValueError(f"invalid move ${player_one}")
        if player_two not in valid:
            raise ValueError(f"invalid move ${player_two}")
        self.player_one = player_one
        self.player_two = player_two

    def winner(self) -> int:
        if self.player_one == self.player_two:
            return 0
        if self.player_one == "rock" and self.player_two == "scissors":
            return -1
        if self.player_one == "scissors" and self.player_two == "rock":
            return 1
        if self.player_one == "paper" and self.player_two == "rock":
            return -1
        if self.player_one == "rock" and self.player_two == "paper":
            return 1
        if self.player_one == "scissors" and self.player_two == "paper":
            return -1
        if self.player_one == "paper" and self.player_two == "scissors":
            return 1
        raise ValueError(f"{self.player_one}:{self.player_two} is undefined")
